0002-logging-Fix-build-with-uClibc.patch depends on this upstream patch.

Signed-off-by: Fatih Aşıcı <fatih.asici@gmail.com>

From d1d3c36e876464a9bae42565f086ded268ab5118 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Wed, 17 Dec 2014 20:24:04 -0800
Subject: [PATCH] Simplify use of __has_include in qlogging.cpp

Easier to just #define it to 0

Change-Id: Ife99fdca6564077762fa67c6d7a5becaf48655d8
Reviewed-by: Olivier Goffart <ogoffart@woboq.com>
---
 src/corelib/global/qlogging.cpp | 21 +++++++++------------
 1 file changed, 9 insertions(+), 12 deletions(-)

diff --git a/src/corelib/global/qlogging.cpp b/src/corelib/global/qlogging.cpp
index 0271573..50d35a6 100644
--- a/src/corelib/global/qlogging.cpp
+++ b/src/corelib/global/qlogging.cpp
@@ -72,20 +72,17 @@
 # include "private/qcore_unix_p.h"
 #endif
 
-#if !defined QT_NO_REGULAREXPRESSION && !defined(QT_BOOTSTRAPPED)
-#ifdef __has_include
-#if __has_include(<cxxabi.h>) && __has_include(<execinfo.h>)
-#define QLOGGING_HAVE_BACKTRACE
-#endif
-#elif defined(__GLIBCXX__) && defined(__GLIBC__) // (because older version of gcc don't have __has_include)
-#define QLOGGING_HAVE_BACKTRACE
+#ifndef __has_include
+#  define __has_include(x) 0
 #endif
 
-#ifdef QLOGGING_HAVE_BACKTRACE
-#include <qregularexpression.h>
-#include <cxxabi.h>
-#include <execinfo.h>
-#endif
+#if !defined QT_NO_REGULAREXPRESSION && !defined(QT_BOOTSTRAPPED)
+#  if (defined(__GLIBC__) && defined(__GLIBCXX__)) || (__has_include(<cxxabi.h>) && __has_include(<execinfo.h>))
+#    define QLOGGING_HAVE_BACKTRACE
+#    include <qregularexpression.h>
+#    include <cxxabi.h>
+#    include <execinfo.h>
+#  endif
 #endif
 
 #include <stdio.h>
-- 
1.9.1

